src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs

```diff
@@ -3055,19 +3055,53 @@ internal static IEdmNavigationSource FindNavigationSourceExtended(this IEdmEntit
         /// <returns>The navigation source found or empty if none found.</returns>
         internal static IEdmNavigationSource FindNavigationSource(this IEdmEntityContainer container, string path)
         {
-            string[] pathSegments = path.Split('.').Last().Split('/');
+            EdmUtil.CheckArgumentNull(container, "container");
+            EdmUtil.CheckArgumentNull(path, "path");
+
+            // the path could be:
+            // NS.Default.Customers/ContainedOrders  --> for back-compatible

```

Comment:
or even NS.Default.Customers (for backward-compatibility) or NS.Default/Customers (for top-level entity set in the Default entity container)